src/seedu/addressbook/commands/Command.java

```diff
@@ -42,6 +42,11 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     public CommandResult execute(){
         throw new UnsupportedOperationException("This method should be implement in child classes");
     }
+    
+    //Returns true for command types that mutate the data, is overridden by commands that are mutating
+    public boolean isMutating(){
+        return false;
```

Comment:
Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.